math/wnaf_test.go

```diff
@@ -35,15 +35,14 @@ func TestOmegaNAF(t *testing.T) {
 
 func TestOmegaNAFRegular(t *testing.T) {
 	testTimes := 1 << 7
-	var max big.Int
-	max.SetInt64(1)
-	max.Lsh(&max, 128)
+	Two128 := big.NewInt(1)
+	Two128.Lsh(Two128, 128)
 
 	for w := uint(2); w < 10; w++ {
 		for j := 0; j < testTimes; j++ {
-			x, _ := rand.Int(rand.Reader, &max)
+			x, _ := rand.Int(rand.Reader, Two128)
 			x.SetBit(x, 0, uint(1)) // odd-numbers
-			L := SignedDigit(x, w)
+			L := SignedDigit(x, w, 128)
```

Comment:
I think we also should test that the length is what we think it is. We might also want to include edge cases in the representation such as 2^w, 2^w-1, etc.